# Business Understanding

**Potential Stakeholders**: WNBA Management, Team Executives, Sports Analysts and Commentators, Agents, Fans

**Stakeholder Actions**:
- **WNBA Management**: Use predictions to inform marketing strategies and policy decisions.
- **Team Executives**: Apply insights to guide roster moves, coaching hires, and playoff preparation.
- **Sports Analysts and Commentators**: Leverage forecasts to enrich game coverage and provide deeper analysis to audiences.
- **Agents**: Identify emerging talent and award contenders for targeted representation and contract negotiations.
- **Fans**: Engage with predictive content to enhance season experience and participate in discussions around team and player performance.

**Available Data**: 10 years of comprehensive WNBA data including player statistics, team performance, coaching records, and awards history. The main risks involve missing values, data inconsistencies, limited sample size, potential class imbalance and integrating the datasets, all of which may affect model accuracy and reliability.

**Business Objectives**:
1. **Improve Team Performance Prediction**: Develop models to accurately predict conference rankings
   - Success Metric: Correctly rank teams within ±2 positions of actual final standings
   - Business Value: Enables better strategic planning for playoffs and roster management

2. **Optimize Coaching Decisions**: Predict which teams are likely to change coaches
   - Success Metric: Identify 70%+ of teams that will undergo coaching changes
   - Business Value: Reduces costs of poor coaching fits and improves team stability

3. **Identify Award Winners Early**: Predict individual award recipients
   - Success Metric: Correctly predict 60%+ of major award winners
   - Business Value: Enhances marketing campaigns and fan engagement throughout season; allows agents and teams to identify promising players to work with

**Secondary Goals**:
- Understand key performance indicators that differentiate successful teams
- Identify player attributes that correlate with award recognition
- Analyze impact of coaching experience on team performance

## Business Goals

### Goal 1: Conference Ranking Prediction
**Data Mining Task**: Regression/Ranking Problem
- **Target Variable**: Team rank within conference
- **Features**: 
  - Team statistics: offensive/defensive efficiency, win%, home/away performance
  - Player statistics: NBA_PER, assist-to-turnover ratio, shooting percentages
  - Historical performance: previous season rankings, playoff appearances
- **Approach**: Ordinal regression or learning-to-rank algorithms
- **Evaluation Metric**: Mean Absolute Error (MAE) in ranking position, Spearman correlation

### Goal 2: Coaching Change Prediction
**Data Mining Task**: Binary Classification Problem
- **Target Variable**: Coach change (Yes/No) in upcoming season
- **Features**:
  - Team performance: win%, playoff participation, win_pct variance
  - Coach statistics: experience (seasons coached), historical win%, recent performance trend
  - Situational factors: consecutive losing seasons, missing playoffs
- **Approach**: Logistic Regression, Random Forest, XGBoost
- **Evaluation Metric**: Precision, Recall, F1-Score (class imbalance likely)

### Goal 3: Award Winner Prediction
**Data Mining Task**: Multi-class Classification Problem (per award category)
- **Target Variables**: 
  - MVP
  - Rookie of the Year
  - Defensive Player of the Year
  - Most Improved Player
- **Features**:
  - Player statistics: NBA_PER, ppg, apg, rpg, spg, bpg, True Shooting %
  - Team context: team win%, playoff success
  - Player position and role
  - Historical award patterns by position
- **Approach**: Multi-class classification (Random Forest, Neural Networks)
- **Evaluation Metric**: Top-3 Accuracy, Precision@K

## Project Plan

**Phase 1: Data Understanding & Preparation**
- Explore and assess data quality across all datasets (teams, players, coaches, awards)
- Handle missing values, outliers, and inconsistencies
- Engineer relevant features for each prediction task
- Create unified dataset with proper temporal alignment

**Phase 2: Model Development**
- Conference Ranking Prediction models
- Coaching Change Prediction models  
- Award Winner Prediction models
- Establish baseline models and iterate with parameter tuning

**Phase 3: Model Evaluation & Selection**
- Compare models against success metrics
- Perform cross-validation and temporal validation
- Select best-performing models for each task

# Data Understanding

## Setup & Data Collection

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from IPython.display import display

In [ ]:
sns.set_theme(style="whitegrid", palette="pastel")
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [ ]:
awards_players_df = pd.read_csv('../datasets/awards_players.csv')
coaches_df        = pd.read_csv('../datasets/coaches.csv')
players_teams_df  = pd.read_csv('../datasets/players_teams.csv')
players_df        = pd.read_csv('../datasets/players.csv')
series_post_df    = pd.read_csv('../datasets/series_post.csv')
teams_post_df     = pd.read_csv('../datasets/teams_post.csv')
teams_df          = pd.read_csv('../datasets/teams.csv')

## Data Description

### *awards_players*

In [19]:
display(awards_players_df.describe(include='all'))
print("\nMissing Values per Column:")
display(awards_players_df.isnull().sum())
print(f"Number of duplicate rows: {awards_players_df.duplicated().sum()}")

,playerID,award,year,lgID
count,95,95,95.000000,95
unique,58,12,NaN,1
top,leslili01w,Coach of the Year,NaN,WNBA
freq,10,10,NaN,95
mean,NaN,NaN,5.789474,NaN
std,NaN,NaN,2.747900,NaN
min,NaN,NaN,1.000000,NaN
25%,NaN,NaN,3.500000,NaN
50%,NaN,NaN,7.000000,NaN
75%,NaN,NaN,8.000000,NaN



Missing Values per Column:


playerID    0
award       0
year        0
lgID        0
dtype: int64

Number of duplicate rows: 0


### *coaches*

In [20]:
display(coaches_df.describe(include='all'))
print("\nMissing Values per Column:")
display(coaches_df.isnull().sum())
print(f"Number of duplicate rows: {coaches_df.duplicated().sum()}")

,coachID,year,tmID,lgID,stint,won,lost,post_wins,post_losses
count,162,162.000000,162,162,162.000000,162.000000,162.000000,162.000000,162.000000
unique,57,NaN,20,1,NaN,NaN,NaN,NaN,NaN
top,donovan99w,NaN,WAS,WNBA,NaN,NaN,NaN,NaN,NaN
freq,9,NaN,13,162,NaN,NaN,NaN,NaN,NaN
mean,NaN,5.314815,NaN,NaN,0.364198,14.672840,14.623457,1.166667,1.172840
std,NaN,2.896715,NaN,NaN,0.693861,6.403445,5.678789,1.953656,1.316782
min,NaN,1.000000,NaN,NaN,0.000000,0.000000,2.000000,0.000000,0.000000
25%,NaN,3.000000,NaN,NaN,0.000000,10.000000,11.000000,0.000000,0.000000
50%,NaN,5.000000,NaN,NaN,0.000000,16.000000,15.000000,0.000000,0.000000
75%,NaN,8.000000,NaN,NaN,0.000000,18.750000,18.000000,1.000000,2.000000



Missing Values per Column:


coachID        0
year           0
tmID           0
lgID           0
stint          0
won            0
lost           0
post_wins      0
post_losses    0
dtype: int64

Number of duplicate rows: 0


### *players_teams*

In [21]:
display(players_teams_df.describe(include='all'))
print("\nMissing Values per Column:")
display(players_teams_df.isnull().sum())
print(f"Number of duplicate rows: {players_teams_df.duplicated().sum()}")

,playerID,year,stint,tmID,lgID,GP,GS,minutes,points,oRebounds,dRebounds,rebounds,assists,steals,blocks,turnovers,PF,fgAttempted,fgMade,ftAttempted,ftMade,threeAttempted,threeMade,dq,PostGP,PostGS,PostMinutes,PostPoints,PostoRebounds,PostdRebounds,PostRebounds,PostAssists,PostSteals,PostBlocks,PostTurnovers,PostPF,PostfgAttempted,PostfgMade,PostftAttempted,PostftMade,PostthreeAttempted,PostthreeMade,PostDQ
count,1876,1876.000000,1876.000000,1876,1876,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000,1876.000000
unique,555,NaN,NaN,20,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,smithka01w,NaN,NaN,PHO,WNBA,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,11,NaN,NaN,152,1876,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,5.326226,0.113539,NaN,NaN,24.320896,12.438166,501.269190,176.261727,24.388060,54.334755,78.722814,39.031983,19.600746,9.065032,36.480277,48.596482,152.122068,64.071962,48.376866,36.358742,34.659382,11.759062,0.415778,1.828358,0.990405,39.950959,14.140725,1.900320,4.413646,6.313966,3.126333,1.420043,0.759062,2.623134,3.735075,12.282516,5.149254,3.672708,2.822495,2.924307,1.019723,0.026652
std,NaN,2.905475,0.422574,NaN,NaN,10.460614,13.641697,359.566117,161.983839,23.325974,48.347088,69.210226,40.147037,17.542694,13.497853,27.956998,34.158825,132.153836,58.914688,48.238245,37.868260,46.189357,17.023107,0.888352,2.659597,2.215079,71.565062,29.551860,4.154121,9.121491,12.881782,7.081885,2.992881,2.280011,5.037807,6.697874,24.313379,10.726421,8.463917,6.723170,7.751034,2.992637,0.170751
min,NaN,1.000000,0.000000,NaN,NaN,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,NaN,3.000000,0.000000,NaN,NaN,17.000000,0.000000,165.000000,41.000000,6.000000,15.000000,21.000000,8.000000,5.000000,1.000000,11.000000,18.000000,42.000000,15.000000,10.000000,7.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,NaN,5.000000,0.000000,NaN,NaN,29.000000,5.000000,459.000000,129.000000,18.000000,42.000000,63.000000,26.000000,16.000000,4.000000,32.000000,47.000000,118.000000,47.000000,33.000000,23.500000,12.000000,4.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,NaN,8.000000,0.000000,NaN,NaN,32.000000,29.000000,826.250000,284.000000,36.000000,83.000000,119.000000,57.250000,30.000000,11.000000,57.000000,74.000000,243.000000,104.000000,73.000000,55.000000,54.000000,18.000000,0.250000,3.000000,0.000000,54.250000,15.250000,2.000000,5.000000,7.000000,3.000000,1.000000,0.000000,3.000000,5.000000,15.000000,6.000000,4.000000,2.000000,1.000000,0.000000,0.000000



Missing Values per Column:


playerID              0
year                  0
stint                 0
tmID                  0
lgID                  0
GP                    0
GS                    0
minutes               0
points                0
oRebounds             0
dRebounds             0
rebounds              0
assists               0
steals                0
blocks                0
turnovers             0
PF                    0
fgAttempted           0
fgMade                0
ftAttempted           0
ftMade                0
threeAttempted        0
threeMade             0
dq                    0
PostGP                0
PostGS                0
PostMinutes           0
PostPoints            0
PostoRebounds         0
PostdRebounds         0
PostRebounds          0
PostAssists           0
PostSteals            0
PostBlocks            0
PostTurnovers         0
PostPF                0
PostfgAttempted       0
PostfgMade            0
PostftAttempted       0
PostftMade            0
PostthreeAttempted    0
PostthreeMade   

Number of duplicate rows: 0


### *players*

In [27]:
display(players_df.describe(include='all'))
print("\nMissing Values per Column:")
display(players_df.isnull().sum())
print(f"Number of duplicate rows: {players_df.duplicated().sum()}")

,bioID,pos,firstseason,lastseason,height,weight,college,collegeOther,birthDate,deathDate
count,893,815,893.0,893.0,893.000000,893.000000,726,11,893,893
unique,893,7,NaN,NaN,NaN,NaN,160,11,764,5
top,abrahta01w,G,NaN,NaN,NaN,NaN,Tennessee,Jefferson College (JC),0000-00-00,0000-00-00
freq,1,309,NaN,NaN,NaN,NaN,38,1,84,889
mean,NaN,NaN,0.0,0.0,65.500560,145.415454,NaN,NaN,NaN,NaN
std,NaN,NaN,0.0,0.0,20.940425,61.275703,NaN,NaN,NaN,NaN
min,NaN,NaN,0.0,0.0,0.000000,0.000000,NaN,NaN,NaN,NaN
25%,NaN,NaN,0.0,0.0,68.000000,140.000000,NaN,NaN,NaN,NaN
50%,NaN,NaN,0.0,0.0,72.000000,162.000000,NaN,NaN,NaN,NaN
75%,NaN,NaN,0.0,0.0,75.000000,180.000000,NaN,NaN,NaN,NaN



Missing Values per Column:


bioID             0
pos              78
firstseason       0
lastseason        0
height            0
weight            0
college         167
collegeOther    882
birthDate         0
deathDate         0
dtype: int64

Number of duplicate rows: 0


### *series_post*

In [23]:
display(series_post_df.describe(include='all'))
print("\nMissing Values per Column:")
display(series_post_df.isnull().sum())
print("\n=== Duplicate Rows ===")
print(f"Number of duplicate rows: {series_post_df.duplicated().sum()}")

,year,round,series,tmIDWinner,lgIDWinner,tmIDLoser,lgIDLoser,W,L
count,70.00000,70,70,70,70,70,70,70.000000,70.000000
unique,NaN,3,7,14,1,18,1,NaN,NaN
top,NaN,FR,A,LAS,WNBA,LAS,WNBA,NaN,NaN
freq,NaN,40,10,12,70,7,70,NaN,NaN
mean,5.50000,NaN,NaN,NaN,NaN,NaN,NaN,2.071429,0.614286
std,2.89302,NaN,NaN,NaN,NaN,NaN,NaN,0.259399,0.572127
min,1.00000,NaN,NaN,NaN,NaN,NaN,NaN,2.000000,0.000000
25%,3.00000,NaN,NaN,NaN,NaN,NaN,NaN,2.000000,0.000000
50%,5.50000,NaN,NaN,NaN,NaN,NaN,NaN,2.000000,1.000000
75%,8.00000,NaN,NaN,NaN,NaN,NaN,NaN,2.000000,1.000000



Missing Values per Column:


year          0
round         0
series        0
tmIDWinner    0
lgIDWinner    0
tmIDLoser     0
lgIDLoser     0
W             0
L             0
dtype: int64


=== Duplicate Rows ===
Number of duplicate rows: 0


### *teams_post*

In [24]:
display(teams_post_df.describe(include='all'))
print("\nMissing Values per Column:")
display(teams_post_df.isnull().sum())
print(f"Number of duplicate rows: {teams_post_df.duplicated().sum()}")

,year,tmID,lgID,W,L
count,80.000000,80,80,80.000000,80.000000
unique,NaN,18,1,NaN,NaN
top,NaN,LAS,WNBA,NaN,NaN
freq,NaN,9,80,NaN,NaN
mean,5.500000,NaN,NaN,2.350000,2.350000
std,2.890403,NaN,NaN,2.228129,0.843441
min,1.000000,NaN,NaN,0.000000,0.000000
25%,3.000000,NaN,NaN,1.000000,2.000000
50%,5.500000,NaN,NaN,1.500000,2.000000
75%,8.000000,NaN,NaN,3.250000,3.000000



Missing Values per Column:


year    0
tmID    0
lgID    0
W       0
L       0
dtype: int64

Number of duplicate rows: 0


### *teams*

In [25]:
display(teams_df.describe(include='all'))
print("\nMissing Values per Column:")
display(teams_df.isnull().sum())
print(f"Number of duplicate rows: {teams_df.duplicated().sum()}")

,year,lgID,tmID,franchID,confID,divID,rank,playoff,seeded,firstRound,semis,finals,name,o_fgm,o_fga,o_ftm,o_fta,o_3pm,o_3pa,o_oreb,o_dreb,o_reb,o_asts,o_pf,o_stl,o_to,o_blk,o_pts,d_fgm,d_fga,d_ftm,d_fta,d_3pm,d_3pa,d_oreb,d_dreb,d_reb,d_asts,d_pf,d_stl,d_to,d_blk,d_pts,tmORB,tmDRB,tmTRB,opptmORB,opptmDRB,opptmTRB,won,lost,GP,homeW,homeL,awayW,awayL,confW,confL,min,attend,arena
count,142.000000,142,142,142,142,0.0,142.000000,142,142.0,80,38,20,142,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.0,142.0,142.0,142.0,142.0,142.0,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142.000000,142
unique,NaN,1,20,18,2,NaN,NaN,2,NaN,2,2,2,20,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,22
top,NaN,WNBA,LAS,CON,WE,NaN,NaN,Y,NaN,L,W,W,Los Angeles Sparks,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Verizon Center
freq,NaN,142,10,10,72,NaN,NaN,80,NaN,40,20,11,10,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,10
mean,5.302817,NaN,NaN,NaN,NaN,NaN,4.084507,NaN,0.0,NaN,NaN,NaN,NaN,860.387324,2039.683099,488.338028,651.366197,157.161972,463.014085,330.500000,730.929577,1061.429577,520.830986,653.929577,263.112676,510.450704,122.077465,2366.274648,860.380282,2039.676056,488.338028,651.366197,157.161972,463.014085,330.514085,730.922535,1061.436620,520.830986,653.915493,263.112676,510.450704,122.070423,2366.260563,0.0,0.0,0.0,0.0,0.0,0.0,16.661972,16.661972,33.323944,10.169014,6.492958,6.492958,10.169014,10.563380,10.563380,6735.683099,141050.253521,NaN
std,2.917274,NaN,NaN,NaN,NaN,NaN,2.095226,NaN,0.0,NaN,NaN,NaN,NaN,86.998969,176.879707,70.749372,86.035246,43.736580,116.166119,41.191432,83.378114,105.393245,54.625738,60.978039,34.912880,50.873585,26.136324,243.154860,82.547277,183.678935,70.896377,90.051245,30.482093,82.147246,33.743659,78.264153,100.099983,50.782106,51.319814,26.644521,54.038019,20.658537,234.615384,0.0,0.0,0.0,0.0,0.0,0.0,4.999131,4.999131,0.949425,2.994017,2.967308,2.702104,2.731409,3.485461,3.485461,197.851093,34714.358519,NaN
min,1.000000,NaN,NaN,NaN,NaN,NaN,1.000000,NaN,0.0,NaN,NaN,NaN,NaN,647.000000,1740.000000,333.000000,469.000000,62.000000,205.000000,242.000000,537.000000,793.000000,390.000000,467.000000,187.000000,408.000000,63.000000,1822.000000,664.000000,1676.000000,325.000000,444.000000,92.000000,315.000000,267.000000,567.000000,873.000000,388.000000,538.000000,197.000000,390.000000,71.000000,1788.000000,0.0,0.0,0.0,0.0,0.0,0.0,4.000000,4.000000,32.000000,1.000000,0.000000,1.000000,3.000000,2.000000,2.000000,6400.000000,57635.000000,NaN
25%,3.000000,NaN,NaN,NaN,NaN,NaN,2.000000,NaN,0.0,NaN,NaN,NaN,NaN,794.500000,1908.500000,435.250000,582.750000,128.250000,389.000000,301.250000,653.250000,969.250000,478.500000,617.000000,241.250000,470.500000,101.250000,2185.250000,800.750000,1889.000000,442.250000,591.250000,135.000000,401.750000,307.250000,676.250000,983.000000,491.250000,616.000000,244.250000,470.250000,109.000000,2196.750000,0.0,0.0,0.0,0.0,0.0,0.0,13.000000,14.000000,32.000000,8.000000,4.250000,5.000000,9.000000,8.000000,8.000000,6500.000000,120897.500000,NaN
50%,5.000000,NaN,NaN,NaN,NaN,NaN,4.000000,NaN,0.0,NaN,NaN,NaN,NaN,864.000000,2025.000000,483.500000,650.000000,157.000000,459.000000,333.500000,724.000000,1069.500000,520.000000,648.000000,261.50000


Missing Values per Column:


year            0
lgID            0
tmID            0
franchID        0
confID          0
divID         142
rank            0
playoff         0
seeded          0
firstRound     62
semis         104
finals        122
name            0
o_fgm           0
o_fga           0
o_ftm           0
o_fta           0
o_3pm           0
o_3pa           0
o_oreb          0
o_dreb          0
o_reb           0
o_asts          0
o_pf            0
o_stl           0
o_to            0
o_blk           0
o_pts           0
d_fgm           0
d_fga           0
d_ftm           0
d_fta           0
d_3pm           0
d_3pa           0
d_oreb          0
d_dreb          0
d_reb           0
d_asts          0
d_pf            0
d_stl           0
d_to            0
d_blk           0
d_pts           0
tmORB           0
tmDRB           0
tmTRB           0
opptmORB        0
opptmDRB        0
opptmTRB        0
won             0
lost            0
GP              0
homeW           0
homeL           0
awayW           0
awayL     

Number of duplicate rows: 0


### Data Description Report

#### Redundant Columns

Across almost all datasets, there are columns containing only a single unique value. These provide no predictive power and should be dropped to reduce dimensionality.

- **League Identifiers**: The lgID column (and variations like lgIDWinner, lgIDLoser) appears in all datasets with the single value "WNBA".

- **Zero-Variance Metrics**: In the *teams* dataset, the following columns contain only zeros (0):

    - seeded

    - tmORB, tmDRB, tmTRB

    - opptmORB, opptmDRB, opptmTRB

    The same happens with these columns from *players*:

    - firstseason, lastseason

#### Missing or Null Values

The *teams* and *players* datasets are affected by missing or null values.

1. *teams*

    - Missing values in firstRound, semis, and finals are not errors but indicators that a team did not advance to those playoff stages.

    - The divID column is entirely empty and provides no value. Therefore, it should be removed.

2. *players*

    - The pos attribute is missing for 78 players.

    - The college and collegeOther columns are missing for a significant number of players (167 and 882 out of 893, respectively). The latter is essentially unusable.

    - There are values of 0 in the weight and height columns, which should be impossible.

    - The most frequent value for birthDate is "0000-00-00" (appearing 84 times), which clearly indicates missing data rather than valid dates. The deathDate column is predominantly "0000-00-00", which is expected for living players.